### ISSUE TITLE
Why expo use/require Location permission? - App rejection in playstore

### ISSUE THREAD

**Issue by AkashWeybee** [expo/expo#11838]
**Summary**

Background location permission issue
My app does not use location or any location related permission.
Why expo require background location access?
Google [playstore] rejecting my app 5 times. Due to this issue, we already exceeded our deadline.

I used to fall in love with the expo but this is a major disappointment.

I must say, this is a major privacy issue.

Please help us.

I didn’t install a single package - Why expo installed a lot of packages without declaring a use case. List of packages - “expo-error-recovery”: “~1.4.0”, “expo-file-system”: “~9.3.0”, “expo-font”: “~8.4.0”, “expo-keep-awake”: “~8.4.0”, “expo-linear-gradient”: “~8.4.0”, “expo-linking”: “~2.0.1”, “expo-location”: “~10.0.0”, “expo-sqlite”: “~8.5.0”.

**Managed or bare workflow? If you have `ios/` or `android/` directories in your project, the answer is bare!**

managed

**What platform(s) does this occur on?**

Android, iOS

**SDK Version (managed workflow only)**

40

**Environment**

  Expo CLI 3.21.5 environment info:
    System:
      OS: Windows 10 10.0.18362
    Binaries:
      Node: 12.16.1 - C:\Program Files\nodejs\node.EXE
      Yarn: 1.22.4 - C:\Program Files (x86)\Yarn\bin\yarn.CMD
      npm: 6.13.4 - C:\Program Files\nodejs\npm.CMD
    IDEs:
      Android Studio: Version  4.1.0.0 AI-201.8743.12.41.6953283

**Reproducible demo or steps to reproduce from a blank project**

Background location permission issue

**Comment by yarapolana** [comment]
@AkashWeybee I think they changed to an opt-in in this version.
Have you tried specifying your permissions in app.json?
```
{ 
  "expo": {
    ...
    "android": {
      "permissions": [
        "CAMERA",
        "READ_CONTACTS",
        ...
      ] 
    }
    ...
  }
}
```

**Comment by byCedric** [comment]
Hi @AkashWeybee, sorry to hear that you are frustrated with this issue. Short answer to your question is: expo doesn't auto-add the background location, but may include foreground location. This is because `expo-location` is part of our main `expo` library, and inherits the permissions _**if you don't specify the [`expo.android.permissions`](https://docs.expo.io/versions/v40.0.0/config/app/#permissions) list**_

If you have excluded the permissions you don't want, but it's still in the build, we can help checking this out. For that, I need your answers to these questions:

1. What is the output of `aapt2 d permissions path/to/app.apk` (See this [installation/usage guide](https://developer.android.com/studio/command-line/aapt2) of aapt2)
2. What is the name of your app (e.g. @acme/app)

**Comment by brentvatne** [comment]
> I didn’t install a single package - Why expo installed a lot of packages without declaring a use case. List of packages - “expo-error-recovery”: “~1.4.0”, “expo-file-system”: “~9.3.0”, “expo-font”: “~8.4.0”, “expo-keep-awake”: “~8.4.0”, “expo-linear-gradient”: “~8.4.0”, “expo-linking”: “~2.0.1”, “expo-location”: “~10.0.0”, “expo-sqlite”: “~8.5.0”.

the expo package currently installs a number of libraries for historical reasons - because we exported them on the `global` expo object. we have deprecated this in sdk 40, and are removing the globals in sdk 41.

**Comment by komelabbbas** [comment]
Same issue...?
Please help.
I have tried many times but fail unfortunately,😓

**Comment by komelabbbas** [comment]
By the way, I have already added specific "permissions": [] in app.json android.

In whole app there is no any location feature but error say that:👇🏻
"Your background location permission declaration needs to be updated."

Any solution plz help.

**Comment by joaops95** [comment]
Hello I'm facing the same issue:
Here's my package.json
`{
  "main": "node_modules/expo/AppEntry.js",
  "scripts": {
    "start": "expo start",
    "android": "expo start --android",
    "ios": "expo start --ios",
    "web": "expo start --web",
    "eject": "expo eject"
  },
  "dependencies": {
    "@expo/plist": "0.0.8",
    "@react-native-community/masked-view": "0.1.10",
    "@sentry/react-native": "^1.4.2",
    "expo": "^40.0.0",
    "expo-apple-authentication": "~2.2.2",
    "expo-cli": "4.1.6",
    "expo-constants": "~9.3.3",
    "expo-facebook": "~9.1.0",
    "expo-notifications": "^0.9.0",
    "expo-updates": "~0.4.1",
    "fast-html-parser": "^1.0.1",
    "html-entities": "^1.3.1",
    "outdarego-api-v3": "^3.1.333",
    "react": "16.13.1",
    "react-dom": "16.13.1",
    "react-native": "https://github.com/expo/react-native/archive/sdk-40.0.1.tar.gz",
    "react-native-fbsdk": "^1.1.2",
    "react-native-gesture-handler": "~1.8.0",
    "react-native-rate": "^1.2.1",
    "react-native-reanimated": "~1.13.0",
    "react-native-responsive-fontsize": "^0.4.3",
    "react-native-safe-area-context": "3.1.9",
    "react-native-screens": "~2.15.2",
    "react-native-svg": "12.1.0",
    "react-native-web": "~0.13.12",
    "react-native-webview": "11.0.0",
    "react-navigation": "^4.3.9",
    "react-navigation-drawer": "^2.4.13",
    "react-navigation-stack": "^2.5.1",
    "sentry-expo": "^2.0.4"
  },
  "devDependencies": {
    "@babel/core": "~7.9.0",
    "babel-preset-expo": "8.3.0"
  },
  "private": true
}
`
The app name is

**Comment by bryanstearns** [comment]
I'm also currently appealing a Play store rejection related to this: our managed app only uses foreground location and geocoding, and the published SDK38-based version had the ACCESS_BACKGROUND_LOCATION permission in its manifest because of `expo-location`. We've upgraded the app to SDK 40 (and bumped the version) and I've verified that the manifest no longer contains that permission, but the Play store rejected the upgraded version - I'm still trying to get them to explain why.

(I'll edit this comment if I have any success.)

**Comment by Denness** [comment]
I had the same issue. I wasn't using many Expo features so I rebuilt the project without expo. When I was resubmitting the project I was still getting location errors even when expo had been removed. I'm new to submitting to the play store so I wasn't sure why I was still getting the error. I managed to get it to work by creating a new app inside the Google Play Store rather than resubmitting it where it was failing before.
